src/libraries/Common/src/Interop/Windows/BCrypt/Interop.BCryptSetProperty.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class BCrypt
+    {
+        [LibraryImport(Libraries.BCrypt, StringMarshalling = StringMarshalling.Utf16)]
+        internal static unsafe partial NTSTATUS BCryptSetProperty(
+            SafeBCryptHandle hObject,
+            string pszProperty,
+            void* pbInput,
+            int cbInput,
+            int dwFlags);
+
+        internal static unsafe void BCryptSetSZProperty(SafeBCryptHandle hObject, string pszProperty, string pszValue)
+        {
+            fixed (void* pbInput = pszValue)
+            {
+                NTSTATUS status = BCryptSetProperty(
+                    hObject,
+                    pszProperty,
+                    pbInput,
+                    (pszValue.Length + 1) * 2,
```

Comment:
This will null-ref if pszValue is null.  Probably worth a Debug.Assert to pinky swear it isn't.

Reply:
I was debating whether it could overflow and convinced myself "no, because .NET object headers limit the max size of a string"... but a) the checked doesn't hurt much, and b) it might since it's currently using signed integers... (b) goes away with the int -> uint above (assuming this changes to 2U), but it still doesn't hurt to put checked in it.